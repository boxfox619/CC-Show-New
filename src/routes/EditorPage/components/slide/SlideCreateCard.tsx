import * as React from 'react';
import styled from 'styled-components';
import { addWhiteIcon } from '../../assets';

const Container = styled.div`
    cursor: pointer;
    position: relative;
    height: 140px;
    margin-top: 20px;
    background: white;
    border-radius: 10px;
    box-shadow: 0 3px 10px 0 rgba(0, 0, 0, 0.26);

    &::before{
    content:'';
    position: absolute;
    left:0;
    top:0;
    width:100%;
    height:100%;
    background-color: rgba(0,0,0,0.2);
    -webkit-animation: fadein 0.4s forwards;
    transition: all 0.5s;
    border-radius: 10px;
    }

    &:hover::before{
    background-color: rgba(0,0,0,0.4);
    }

    & > img{
        position: absolute;
        width: 5em;
        height: 5em;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
    }
`
const Description = styled.div`
    position: absolute;
    text-align: center;
    font-weight: gray;
    font-size: 15px;
    color: white;
    width: 100%;
    bottom: 15%;
`

interface Props {
    onClick: () => void,
}

const SlideCreateCard: React.FC<Props> = (props: Props) => {
    return (
        <Container onClick={props.onClick}>
            <Description>새 슬라이드</Description>
            <img src={addWhiteIcon} alt="Add new slide" />
        </Container>
    )
}

export default SlideCreateCard;