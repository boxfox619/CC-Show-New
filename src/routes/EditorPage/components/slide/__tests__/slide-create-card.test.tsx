import * as React from 'react';
import * as Enzyme from 'enzyme';
import SlideCreateCard from './../SlideCreateCard';

describe('<SlideCreateCard />', () => {
  const mockFn = jest.fn();
  it('should match snapshot', () => {
    const wrapper = Enzyme.shallow(<SlideCreateCard onClick={mockFn}/>);
    expect(wrapper).toMatchSnapshot();
  });
});