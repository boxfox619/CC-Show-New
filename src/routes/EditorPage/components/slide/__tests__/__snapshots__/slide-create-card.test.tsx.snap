// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideCreateCard /> should match snapshot 1`] = `ShallowWrapper {}`;
